src/NHibernate.Test/CfgTest/LoadVirtualAssembyTest.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+using NHibernate.Cfg;
+using System.CodeDom.Compiler;
+using Microsoft.CSharp;
+using System.Reflection;
+using NHibernate.Mapping.ByCode;
+using NHibernate.Mapping.ByCode.Conformist;
+
+namespace NHibernate.Test.CfgTest
+{
+	[TestFixture]
+	class LoadVirtualAssembyTest
+    {
+
+		[Test]
+		public void VirtualAssembyLoadTest()
+		{
+			int assemblyGenerateCount = 10;
+			var config = new Configuration();
+			var mapper = new ModelMapper();
+			var module = Compile(assemblyGenerateCount);
+			config.AddMemoryAssembly(module.Assembly);
+			mapper.AddMappings(module.Assembly.GetExportedTypes());
+			config.AddMapping(mapper.CompileMappingForAllExplicitlyAddedEntities());
+
+			if (assemblyGenerateCount == config.ClassMappings.Count)
+			{
+				Assert.Pass("All virtual assembly mapped.");
+			}
+			else
+			{
+				Assert.Fail("Not all virtual assembly mapped.");
+			}
+		}
+
+
+		public Module Compile(int assemblyCount = 1)
+		{
+			CompilerParameters CompilerParams = new CompilerParameters();
+			CompilerParams.GenerateInMemory = true;
+			CompilerParams.TreatWarningsAsErrors = false;
+			CompilerParams.GenerateExecutable = false;
+			CompilerParams.CompilerOptions = "/optimize";
+			CompilerParams.ReferencedAssemblies.AddRange(new String[] {
+				"System.dll",
+                "System.Core.dll",
+                "System.Data.dll",
+				"C:\\NHibernate.dll" //load compiled version of NHibernate
```

Comment:
Not really portable, thus the CI failures. Can't you load from assembly containing a NHibernate type?